### DISCUSSION TITLE
How to test a plugin during development?

### DISCUSSION THREAD

**Discussion by PEZ** [prettier/prettier-vscode#2781] [Q&A]
Hi! Thanks for this Extension!

I am creating a Prettier plugin (for Clojure). I can run prettier just fine from the command line, and it formats Clojure files as well as `clojure` fenced code blocks in Markdown. However, I fail to tell VS Code that the Prettier Code Formatter extension now can format Clojure files. I've read these:

* #395
* #899

Frome those I seem to conclude that if I `npm install` prettier and my plugin, then this extension should pick it up. I still get this, though:

<img width="329" alt="image" src="https://user-images.githubusercontent.com/30010/200518977-52e8f48b-5b83-4705-9042-5b23aa4a315a.png">

I don't know what to try now. Any ideas?

**Comment by nelson6e65** [comment]
Have you tried to set the `editor.defaultFormatter` option?

`settings.json`:
```json
{
  "editor.defaultFormatter": "esbenp.prettier-vscode"
}
```

You can also wrap it for a specific language:
```json
{
  "[dockercompose]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
}
```


But you have to `npm add -D prettier` in your project (make sure to use the correct version here, as version 3 is not compatible with some plugins) and use a local `.prettierrc.json`.

Also, `npm link` in your local package and `npm link clojure-prettier-plugin` (replace the name of your local package) in your example project you want to use to test your plugin.

In your `.prettierrc.json` you need to add your plugin as `{ "plugins": ["clojure-prettier-plugin"] }`.
